packages/seal-mongodb-adapter/src/MongoDBSchemaManager.php

```diff
@@ -0,0 +1,158 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the CMS-IG SEAL project.
+ *
+ * (c) Alexander Schranz <alexander@sulu.io>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace CmsIg\Seal\Adapter\MongoDB;
+
+use CmsIg\Seal\Adapter\SchemaManagerInterface;
+use CmsIg\Seal\Schema\Field;
+use CmsIg\Seal\Schema\Index;
+use CmsIg\Seal\Task\SyncTask;
+use CmsIg\Seal\Task\TaskInterface;
+use MongoDB\Response\MongoDB;
+
+final class MongoDBSchemaManager implements SchemaManagerInterface
+{
+    public function __construct(
+        private readonly ClientWrapper $client,
+    ) {
+    }
+
+    public function existIndex(Index $index): bool
+    {
+        $collectionNames = [...$this->client->getDatabase()->listCollectionNames()];
+
+        return \in_array($index->name, $collectionNames, true);
+    }
+
+    public function dropIndex(Index $index, array $options = []): TaskInterface|null
+    {
+        $this->client->getDatabase()->dropCollection($index->name);
+        $this->client->getDatabase()->getCollection($index->name)->dropSearchIndex($index->name . '-search-index');
+
+        if (!($options['return_slow_promise_result'] ?? false)) {
+            return null;
+        }
+
+        return new SyncTask(null); // TODO wait for index drop
```

Comment:
Here is the code I use to pull the server until the index is ready: https://github.com/mongodb/mongo-php-library/blob/ae6a3a53d507fe38e447dfde4e77c8efe5724925/tests/SpecTests/SearchIndexSpecTest.php#L54